flink-filesystems/flink-s3-fs-base/pom.xml

```diff
@@ -231,6 +231,33 @@ under the License.
 					<skip>true</skip>
 				</configuration>
 			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
```

Comment:
it would be safer to move this into the presto/hadoop modules; as it stands you're excluding one dependency (and kicking out a transitive one), with downstream modules not being aware that this module is bundling anything and hence also trying to bundle `aws-java-sdk-s3`.

This is the warning you get when building downstream modules:
```
[WARNING] flink-s3-fs-base-1.11-SNAPSHOT.jar, aws-java-sdk-s3-1.11.271.jar define 29 overlapping classes: 
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$GetObjectTaggingHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListPartsHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$BucketLocationHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListVersionsHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListAllMyBucketsHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketInventoryConfigurationsHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$AccessControlListHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$InitiateMultipartUploadHandler
[WARNING]   - com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$GetBucketInventoryConfigurationHandler
[WARNING]   - 19 more...
```

As it stands this will require duplicating things; it would've easier if the S3 filesystems where organized in a hierarchy such that we could use inheritance for this stuff.

Reply:
I will try to pursuit the hierarchy approach.

Reply:
Unfortunately, it's not that easy. If it's a parent project, it needs to be of type pom and cannot ship code by default.
There is a [workaround](https://stackoverflow.com/questions/12305190/how-to-generate-jar-for-maven-parent-project), but it's certainly adding some complexity.
The cleanest solution would be to add another module just for the pom, but that's also more complex than I'd like it to be.

Reply:
it doesn't have to ship code; you can define the shade plugin configuration in the parent, which is then inherited by the child modules, which then do the inclusion/exclusion.

Reply:
Not sure what you mean but `flink-s3-fs-base` has code that is lost when I change packaging to `pom`, which in turn is required for parent projects.

Reply:
I forgot that S3 base contains code, ignore the suggestion about parent modules.